CompilerSource/parsing/parser.cpp

```diff
@@ -1494,11 +1494,8 @@ std::unique_ptr<AST::Node> AstBuilder::TryParseOperand() {
   switch (token.type) {
     case TT_BEGINBRACE: case TT_ENDBRACE:
     case TT_ENDPARENTH: case TT_ENDBRACKET:
-    case TT_ENDOFCODE:
+    case TT_ENDOFCODE: case TT_SEMICOLON:
       return nullptr;
-
-    case TT_SEMICOLON:
-      return std::make_unique<AST::CodeBlock>();
     case TT_COLON:
```

Comment:
I need to know if it found `;` so I can handle cases like `x++++;`. This can be determined by the `nullptr`.

Reply:
Possibly dangerous, or possibly obviously correct depending on how TryParseOperand is being used. Either way, good change. Be aware:

```cc
if (condition); else { do_something(); }
```

This is valid in C++. The `;` ends the `if`'s true branch, meaning `else` can happen immediately. I don't really care about this behavior in ENIGMA, but failing to accept it could cause misbehavior. From the name of this function, I don't think this case broke.

Reply:
Nice case, but I think it needs to be handled in `TryParseStatement`.